server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -177,6 +180,11 @@ public MetadataCreateIndexService(
         this.forbidPrivateIndexSettings = forbidPrivateIndexSettings;
         this.shardLimitValidator = shardLimitValidator;
         this.awarenessReplicaBalance = awarenessReplicaBalance;
+
+        /*
+         * Task will get retried from { @link TransportClusterManagerNodeAction}
+         */
+        ClusterManagerTaskThrottler.registerThrottlingKey(CREATE_INDEX_CLUSTER_MANAGER_TASK_KEY, true);
     }
```

Comment:
1. Can we move the key's definition too in the `ClusterManagerTaskThrottler` class
2. Can we pass ClusterManagerTaskThrottler#registerThrottlingKey as a `Consumer` to this class, something like
```
private BiConsumer<String, Boolean> taskThrottlingRegistration
```

and then

```
taskThrottlingRegistration.accept(ClusterManagerTaskThrottler.CREATE_INDEX_CLUSTER_MANAGER_TASK_KEY, true);
```

Reply:
We are planing to expose the registering functionality via cluster service only, do you think we still need consumer?  As it will be single and only way for registering throttling keys. 

All the services has access to cluster service, so they can register it via cluster service. If we want to introduce the consumer, we will need to change constructor of almost all services to have this argument. 

Thoughts?